mlir/lib/IR/Operation.cpp

```diff
@@ -1390,6 +1390,14 @@ LogicalResult OpTrait::impl::verifyIsIsolatedFromAbove(Operation *isolatedOp) {
   return success();
 }
 
+LogicalResult OpTrait::impl::verifyIndexResultType(Operation *op) {
+  if (op->getNumResults() != 1)
+    op->emitError("operation's result number should be 1.");
```

Comment:
It's not clear why this operation should always generate one result - why not have the ability to generate multiple results with all of those being valid symbols?

Reply:
I'm a little less sure there's such a thing, at first I thought multiple results and those results were a little complicated, but I think it's fine.